modules/root/protected_root_api.rb

```diff
@@ -0,0 +1,42 @@
+class Proxy::ProtectedRootApi < Sinatra::Base
+  helpers ::Proxy::Helpers
+
+  # authorize_with_trusted_hosts
+  # authorize_with_ssl_client
+
+  get "/v2/features" do
+    begin
+      enabled_plugins = ::Proxy::Plugins.instance.select do |p|
+        p[:name] != :foreman_proxy && \
+          p[:state] == :running && \
+          p[:class].ancestors.include?(::Proxy::Plugin)
+      end
+      content_type :json
+
+      attributes = %i[http_enabled https_enabled settings]
+
+      plugins = enabled_plugins.inject({}) do |hash, plugin|
+        result = plugin.slice(*attributes)
+        result[:capabilities] = process_capabilities(plugin[:capabilities])
+        hash[plugin[:name]] = result
+        hash
+      end
+
+      plugins.to_json
+    rescue => e
+      log_halt 400, e
```

Comment:
Should this be an internal server error?

Reply:
This was code you wrote :) but yes i think you are correct!

Reply:
I thought I could pin it on you now ;) But yes, my was very much WIP